lua/entities/sent_spawnpoint/init.lua

```diff
@@ -29,18 +34,18 @@ function unlinkPlayerFromSpawnPoint( player, spawnPoint )
 end
 
 function unlinkAllPlayersFromSpawnPoint( spawnPoint, excludedPlayers )
-    if ( IsValid(spawnPoint) ) then
-        local linkedPlayers = spawnPoint.LinkedPlayers
-        local spawnPointOwner = spawnPoint:CPPIGetOwner()
-
-        for player, _ in pairs( linkedPlayers ) do
-            local playerIsExcluded = excludedPlayers[player]
-            local playerOwnsSpawnPoint = spawnPointOwner == player
-			
-            if ( not playerIsExcluded and not playerOwnsSpawnPoint ) then
-                unlinkPlayerFromSpawnPoint( player, spawnPoint )
-                player:PrintMessage(4, "You've been unlinked from a Spawn Point!")
-            end
+    if not IsValid( spawnPoint ) then return end
+
+    local linkedPlayers = spawnPoint.LinkedPlayers
+    local spawnPointOwner = spawnPoint:CPPIGetOwner()
+    
+    for player, _ in pairs( linkedPlayers ) do
```

Comment:
how early and how often are we talking here

```lua
for player, _ in pairs( linkedPlayers ) do
    if excludedPlayers[player] then next end
    if spawnPointOwner == player then next end

    unlinkPlayerFromSpawnPoint( player, spawnPoint)
    player:PrintMessage(4, "You've been unlinked from a Spawn Point!")
end
```

Reply:
`next` does not work like we want it to